torch_harmonics/convolution.py

```diff
@@ -218,35 +230,37 @@ def _precompute_convolution_tensor_s2(
         phi = torch.where(phi < 0.0, phi + 2 * torch.pi, phi)
 
         # find the indices where the rotated position falls into the support of the kernel
-        iidx, vals = filter_basis.compute_support_vals(theta, phi, r_cutoff=theta_cutoff_eff)
+        iidx, vals = filter_basis.compute_support_vals(theta, phi, r_cutoff=theta_cutoff_eff, support_only=support_only)
 
         # add the output latitude and reshape such that psi has dimensions kernel_shape x nlat_out x (nlat_in*nlon_in)
         idx = torch.stack([iidx[:, 0], t * torch.ones_like(iidx[:, 0]), iidx[:, 1] * nlon_in + iidx[:, 2]], dim=0)
 
-        # append indices and values to the COO datastructure
+        # append indices and values to the COO datastructure, compute row offsets
         out_idx.append(idx)
         out_vals.append(vals)
+        out_roff[t + 1] = out_roff[t] + iidx.shape[0]
```

Comment:
what is this for?

Reply:
ah nevermind - just saw it. Did you check that the computation of roff is correct and consistent with what we had before?